src/rpc/client.rs

```diff
@@ -153,7 +153,7 @@ impl Debug for UrlClient {
 
 impl UrlClient {
     async fn new(url: Url, token: impl Into<Option<String>>) -> Result<Self, ClientError> {
-        let timeout = Duration::MAX; // we handle timeouts ourselves.
```

Comment:
Will a subscription method like `ChainNotify` behave correctly with this change?

Reply:
Right now, we don't have a clear plan for testing such methods, so we ignore it.

Reply:
Why do we ignore it? I understand we don't have an automated way of testing this yet. If you set locally the timeout to 10s, will this method still work as expected after 15s?

Reply:
Also, why does the overflow occur? Why is a value added to the timeout we don't use?

Reply:
> Why do we ignore it?

Testing a method like `ChainNotify` and `EthSubscribe` requires new infrastructure code that we don't have at the moment.

Reply:
> If you set locally the timeout to 10s, will this method still work as expected after 15s?

Which method are you referring to?

Reply:
I'm referring to `ChainNotify`. We implemented it, so we had to test it somehow, right?